pkg/scalers/cpu_memory_scaler.go

```diff
@@ -0,0 +1,101 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"k8s.io/api/autoscaling/v2beta2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+type cpuMemoryScaler struct {
+	metadata     *cpuMemoryMetadata
+	resourceName v1.ResourceName
+}
+
+type cpuMemoryMetadata struct {
+	Type               v2beta2.MetricTargetType
+	Value              *resource.Quantity
+	AverageValue       *resource.Quantity
+	AverageUtilization *int32
+}
+
+// NewCPUMemoryScaler creates a new cpuMemoryScaler
+func NewCPUMemoryScaler(resourceName v1.ResourceName, resolvedEnv, metadata map[string]string) (Scaler, error) {
+	meta, parseErr := parseResourceMetadata(metadata, resolvedEnv)
+	if parseErr != nil {
+		return nil, fmt.Errorf("error parsing cron metadata: %s", parseErr)
+	}
+
+	return &cpuMemoryScaler{
+		metadata:     meta,
+		resourceName: resourceName,
+	}, nil
+}
+
+func parseResourceMetadata(metadata, resolvedEnv map[string]string) (*cpuMemoryMetadata, error) {
+	meta := &cpuMemoryMetadata{}
+	if val, ok := metadata["type"]; ok && val != "" {
+		meta.Type = v2beta2.MetricTargetType(val)
+	} else {
+		return nil, fmt.Errorf("no type given")
+	}
+
+	var value string
+	var ok bool
+	if value, ok = metadata["value"]; !ok || value == "" {
+		return nil, fmt.Errorf("no value given")
+	}
+	switch meta.Type {
+	case v2beta2.ValueMetricType:
+		valueQuantity := resource.MustParse(value)
+		meta.Value = &valueQuantity
+	case v2beta2.AverageValueMetricType:
+		averageValueQuantity := resource.MustParse(value)
+		meta.AverageValue = &averageValueQuantity
+	case v2beta2.UtilizationMetricType:
+		valueNum, err := strconv.Atoi(value)
+		if err != nil {
+			return nil, err
+		}
+		utilizationNum := int32(valueNum)
+		meta.AverageUtilization = &utilizationNum
+	default:
+		return nil, fmt.Errorf("unsupport type")
+	}
+	return meta, nil
+}
+
+// IsActive always return true for cpu/memory scaler
```

Comment:
This would cause the target Deployment to be always active eg. scaled to 1. It will not scale to 0.

Reply:
cpu and memory resources are the basic resources of the application. I think IsActive is always true.

In which case do you think false should be returned. do you have any suggestions?

/cc @tomkerkhove

Reply:
I agree, but we should be vocal about this in the scalers documenation. To notice, that these scalers will never scale to 0 and even when user define multiple scaler types (eg. Kafka + cpu/memory, or Prometheus + cpu/memory), the deployment will never scale to 0

Reply:
No problem, I should add more instructions to the document.  Thanks a lot

Reply:
I have added a hint in the document.  docs pr :https://github.com/kedacore/keda-docs/pull/277